### PR TITLE
Add utility methods for parsing and formatting multivalued cell values

### LINKED ISSUE #375: Multivalued value formatting changes after validation
Steps to reproduce:

1. Load the development interface with the CanCOGeN Covid-19 template. Scroll over to the "anatomical part" column.
2. Using the multivalued picker, choose a few values, and click "Ok" in the modal.
3. Note at this point the value is formatted in the cell as the values you chose separated by `; `, for example "Eye; Intestine".
4. Click the "Validate" button.
5. Note that now the value in the cell is formatted as delimited by `;` (no space), for example "Eye;Intestine"

This causes issues because multivalued values don't survive a cycle of enter data -> validate -> export as data objects -> load data objects. 

From a technical point of view there are a number of places where we do a `split` or `join` on multivalued values, but they often differ just a bit in how they treat whitespace. I'll put together a PR to introduce utility methods to normalize parsing and formatting multivalued values.

### PR DESCRIPTION
**Pull request by pkalita-lbl**
These changes introduce two complimentary utility methods, `parseMultivaluedValue` and `formatMultivaluedValue`. They are used during multivalued cell editing, data object loading, and validation to normalize the behavior of translating between arrays and delimited strings. See accompanying tests for details about how they handle edge cases. 

Fixes #375

### CODE REVIEW

File: lib/utils/fields.js [hunk 1]
```diff
@@ -89,12 +83,48 @@ export function dataObjectToArray(dataObject, fields, options = {}) {
...
+export function parseMultivaluedValue(value) {
+  if (!value) {
+    return [];
+  }
+  // trim the delimiter and the resulting tokens to be flexible about what
+  // this function accepts
+  return value
+    .split(MULTIVALUED_DELIMITER.trim())
+    .map((v) => v.trim())
+    .filter((v) => !!v);
```
Comment (ddooley) [inline lib/utils/fields.js:112]: Just curious, does ".filter((v) => !!v);" remove null values?

File: lib/utils/fields.js [hunk 2]
```diff
@@ -89,12 +83,48 @@ export function dataObjectToArray(dataObject, fields, options = {}) {
...
+ * @return {String} String-formatted multivalued value.
+ */
+export function formatMultivaluedValue(values) {
+  if (!values) {
+    return '';
+  }
+
+  return values
+    .filter((v) => !!v)
+    .map((v) => (typeof v === 'string' ? v.trim() : String(v)))
```
Comment (ddooley) [inline lib/utils/fields.js:128]: And this trims strings, and converts numeric to string?

**Review by ddooley — APPROVED** [review]

File: lib/utils/fields.js (hunk 1 above)
Comment (pkalita-lbl) [reply lib/utils/fields.js:112]: It filters out all [falsy values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) by casting the value to a boolean (`!v`) and then inverting the result (`!!v`). So yeah here it's being used to filter out nulls and empty strings.

File: lib/utils/fields.js (hunk 2 above)
Comment (pkalita-lbl) [reply lib/utils/fields.js:128]: Correct. Technically anything that's not a string will get converted to a string, but yeah it's primary to handle numbers here.
